lib/api/CFieldDataCategorizer.cc

```diff
@@ -469,6 +469,10 @@ bool CFieldDataCategorizer::doPersistState(const TStrVec& partitionFieldValues,
                                            const TPersistFuncVec& dataCategorizerPersistFuncs,
                                            std::size_t categorizerAllocationFailures,
                                            core::CDataAdder& persister) {
+
+    // Takes care of clearing the cache of program counters when exiting the current scope.
+    core::CProgramCounters::CCacheManager cacheMgr;
```

Comment:
This is here for the case of the standalone `categorizer` program right?

I think it's unnecessary in the case of categorization state being persisted as part of a joint categorization/anomaly detection job run by `autodetect`.

For the standalone `categorize` program there is a much bigger problem in that the counters will never get cached or persisted at all.  It's not a problem at the moment as development of the pure standalone categorization program was halted - it's impossible to use it in production.  Therefore I think it would be best to remove this line and replace it with a comment:

```suggestion
    // TODO: if the standalone categorize program is ever progressed, a mechanism needs
    // to be added that does the following:
    // 1. Caches program counters in the foreground before starting background persistence
    // 2. Calls core::CProgramCounters::staticsAcceptPersistInserter once and only once per persist
    // 3. Clears the program counter cache after persistence is complete
```